### PR TITLE
Post Content Link Tracking

### PR DESCRIPTION
**Pull request by ItzNotABug**
This PR adds the ability to track URL clicks in the email. We use a redirect logic to do the tracking.

**What links are tracked -**
1. All links directing to sites outside the Ghost domain are fully tracked, providing insights into user interaction with external content.

2. Partial support for for Internal Ghost Domain Links
    - **Excluded:** 
        - URL of/in the Featured Image and its caption (if found).
        - URLs that navigate the user to current post (View in browser), comments section, subscription page, unsubscribe page.
    - **Supported:** 
        - Inter-article links, including those within the body of the article and in the "Keep Reading" section, are fully supported and tracked.

---
### Limitation -
URLs are tracked using a redirect logic so if your server hosting `Ghosler` goes down or is not available when someone clicks a URL, the redirection would fail & the URL won't go to the actual destination.

### CODE REVIEW

File: views/partials/settings/customise.ejs [hunk 1]
```diff
@@ -7,47 +7,61 @@
...
                     <% } %>
             >
         </div>
 
-        <div class="form-group-row ">
+        <div class="form-group-row">
+            <label>Track URL Clicks
+                <span style="font-weight: normal; font-size: 12px; color: #738a94">
+                    <!-- Linking a future PR, hehe. -->
+                    (<a href="https://github.com/itznotabug/ghosler/pull/7" target="_blank"
```
Comment (ItzNotABug) [inline views/partials/settings/customise.ejs]: Should be `8`, guess we used `7` while testing!

**Comment by ItzNotABug** [comment]
An updated `README.md` highlighting this feature would also be a good idea I guess.

File: views/partials/settings/customise.ejs (hunk 1 above)
Comment (ItzNotABug) [reply views/partials/settings/customise.ejs]: Done, removed the comment since its no longer a future PR.

File: views/dashboard/analytics.ejs [hunk 2]
```diff
@@ -245,6 +245,14 @@
                     </td>
                     <td data-title="Newsletter Status"><%= element.stats.newsletterStatus; %></td>
                     <td data-title="Action">
+                        <% if (element.stats.postContentTrackedLinks.length > 0) { %>
```
Comment (ItzNotABug) [inline views/dashboard/analytics.ejs]: This will cause an issue as it wasn't present in the previous versions. A safety check is necessary.

Comment (ItzNotABug) [reply views/dashboard/analytics.ejs]: Done.

**Comment by emanborg** [comment]
I'm hosting ghosler and mailserver in intranet, so newsletters are being sent using the server's internal IP instead. (the tracking url ends up with http://192.168.100.20 instead of the public url. Shall I edit the config file manually using external IP or you suggest something else?

**Pull request merged**